src/lib/components/toast/Toast.component.tsx

```diff
@@ -0,0 +1,180 @@
+import { motion } from 'framer-motion';
+import React, { ReactNode, useRef } from 'react';
+import { useTheme } from 'styled-components';
+import { Box } from '../box/Box';
+import { Button } from '../buttonv2/Buttonv2.component';
+import { Icon, IconColor } from '../icon/Icon.component';
+import { BasicText } from '../text/Text.component';
+import { ToastPosition, positionOutput } from './ToastPositionHelpers';
+import { ToastProgressBar } from './ToastProgressBar';
+import { useToastParameters } from './useToastParameters';
+import styled from 'styled-components';
+
+export type ToastStatus = 'success' | 'error' | 'warning' | 'info';
+
+export type ToastProps = {
+  open: boolean;
+  message: ReactNode;
+  onClose: () => void;
+  status?: ToastStatus;
+  position?: ToastPosition;
+  autoDismiss?: boolean;
+  duration?: number;
+  icon?: React.ReactNode;
+  width?: React.CSSProperties['width'];
+  withProgressBar?: boolean;
+  style?: React.CSSProperties;
+};
+
+export const useGetBackgroundColor = (status: string) => {
+  const theme = useTheme();
+  switch (status) {
+    case 'success':
+      return theme.statusHealthy;
+    case 'error':
+      return theme.statusCritical;
+    case 'warning':
+      return theme.statusWarning;
+    default:
+      return theme.infoPrimary;
+  }
+};
+
+const useGetRgbBackgroundColor = (status: string) => {
+  const theme = useTheme();
+  switch (status) {
+    case 'success':
+      return 'rgba(10, 173, 166, 0.4)';
```

Comment:
nit: I wonder if those rgba shouldn't either be added to the theme either be derived from one of the theme variable

Reply:
opacity can vary from case to case in my opinion, but the rgb value doesn't, so maybe just add the RGB value?

Reply:
Agree !